### ISSUE TITLE
Need of a testing utility for SQL parser of OpenLineage

### ISSUE THREAD

**Issue by howardyoo** [OpenLineage/OpenLineage#997]
SQL Parser library of OpenLineage is a crucial package that gets used frequently in many database oriented extractors. Many customers also develop their own custom extractors, and uses openlineage's sql parser package in them.

A usual way to develop and test customer's custom extractor is by running those extractors on top of airflow and troubleshooting it. However, this introduces many inefficiencies if customers are simply trying to test how the sql parser package is parsing their SQL statements - and whether the parsing is working correctly.

In order to better troubleshoot parsing of SQL statements, there is a need to have a utility that can 
1. Parse the SQL statement only and derive the SQL metadata
2. check whether there hasn't been any errors during the parsing scheme, as well as check the generated metadata output to ensure proper data are extracted and present.

Therefore, I would like to propose a 'sql test' utility which can:
1. input any arbitrary SQL statement in string format.
2. generate sql metadata out of it.
3. print the content of the metadata to console.

This will greatly help reduce the development and test cycle of users who are either working with any SQL related extractors for openlineage.

**Comment by mobuchowski** [comment]
Hey @howardyoo why do we need that in repository? 

Right now to do what you want, you can either 

1. write tests in Rust
2. write tests in Python
3. directly import `openlineage_sql` from pypi and write any test code you want. 

What is insufficient in those approaches?

**Comment by howardyoo** [comment]
Hey @mobuchowski,
So, basically, I do receive frequent request from users who develop custom extractor/operator in airflow who constantly ask me to verify whether their SQLs that goes through their operator actually does work (openlineage_sql)

1. write tests in Rust - It is possible to update test cases on every one of their requests, but I often found it to be cumbersome and repetitious. Besides, their work is mostly on custom extractors, so there isn't easy way to make it as part of the test.
2. Same reason with #1, their custom work also hinders of making this into unit tests.
3. I can definitely suggest 'hey, why don't you write the test code you want by yourself?' but, once again, I don't think asking that to every users who has the same need is an effective solution.

After a while, I found out that what they really wanted was a test script or utility that can simply run the 'parse' part of the openlineage_sql, and see if the statement is properly parsed into the expected metadata (in and out table meta). They may have varying degree of SQLs that they want to test, and something that they can repeatedly perform on many ad-hoc queries without having to update our unit tests. The PR that I made would work great, and I wanted to include it in the repo as part of what we provide back to the users, so that they can run the tests by themselves.
